Zend/tests/pipe_operator/mixed_callable_call.phpt

```diff
@@ -0,0 +1,62 @@
+--TEST--
+Pipe operator handles all callable styles
+--FILE--
+<?php
+
+function _add(int $x, int $y): int {
+    return $x + $y;
+}
+
+function _area(int $x, int $y): int {
+    return $x * $y;
+}
+
+class _Test
+{
+    public function message(string $which): string
+    {
+        if ($which == 1) {
+            return "Hello";
+        }
+        else if ($which == 2) {
+            return "Goodbye";
+        }
+        else {
+            return "World";
+        }
+    }
+}
+
+class StaticTest {
+    public static function oneMore(int $x): int {
+        return $x + 1;
+    }
+}
+
+function _double(int $x): int {
+    return $x * 2;
+}
+
+function multiplier(int $x): \Closure
+{
+    return fn($y) => $x * $y;
+}
+
+$test = new _Test();
+
+$add3 = fn($x) => _add($x, 3);
+
+$res1 = 2
+    |> [$test, 'message']
```

Comment:
We're missing `$test->message(...)`, `[StaticTest::class, 'oneMore']`, `StaticTest::class . '::' . 'oneMore'` and possibly an invokable object if we want to be comprehensive.